integration-cli/docker_cli_build_test.go

```diff
@@ -734,30 +680,23 @@ func (s *DockerSuite) TestBuildCopyWildcardInName(c *check.C) {
 
 func (s *DockerSuite) TestBuildCopyWildcardCache(c *check.C) {
 	name := "testcopywildcardcache"
-	ctx, err := fakeContext(`FROM busybox
+	ctx := fakeContext(c, `FROM busybox
 	COPY file1.txt /tmp/`,
 		map[string]string{
 			"file1.txt": "test1",
 		})
-	if err != nil {
-		c.Fatal(err)
-	}
 	defer ctx.Close()
 
-	id1, err := buildImageFromContext(name, ctx, true)
-	if err != nil {
-		c.Fatal(err)
-	}
+	buildImageSuccessfully(c, name, withExternalBuildContext(ctx))
+	id1 := getIDByName(c, name)
```

Comment:
Also not for this PR, but instead of doing an _inspect_ for each image we build, why don't we capture the output of `docker build -q -t foo .`?

If we have `buildImageSuccessfully` return a slightly different result (e.g. `BuildResult`) (also see my other suggestion) that contains the image-id, we can skip all these